pandas/tseries/frequencies.py

```diff
@@ -298,6 +301,50 @@ def to_offset(freqstr):
             name, stride = stride, name
         name, _ = _base_and_stride(name)
         delta = get_offset(name) * stride
+
+    elif isinstance(freqstr, tslib.Timedelta):
+        name_to_offset_map = {'days': Day(1),
+                              'hours': Hour(1),
+                              'minutes': Minute(1),
+                              'seconds': Second(1),
+                              'milliseconds': Milli(1),
+                              'microseconds': Micro(1),
+                              'nanoseconds': Nano(1)}
+        try:
+            for name in freqstr.components._fields:
+                offset = name_to_offset_map[name]
+                stride = getattr(freqstr.components, name)
+                if stride is not None and int(stride) != 0:
+                    offset = int(stride) * offset
+                    if delta is None:
+                        delta = offset
+                    else:
+                        delta = delta + offset
+        except Exception, ex:
+            raise ValueError("Error evaluating %s. Reason: %s" % (freqstr, ex))
+
+    elif isinstance(freqstr, timedelta):
```

Comment:
done. agreed and this way to_offset will be more resilient to datetime.timedelta evolution.